### PR TITLE
[bsdfs/roughconductor] Include specular_reflectance component

### PR DESCRIPTION
**Pull request by jammm**
Some scenes (such as glass-of-water) require a custom specular_reflectance component to be set.
As this is my first PR, please let me know if I made any mistake :-)

### CODE REVIEW

**Comment by wjakob** [comment]
Hi @jammm,

this looks good to me, but I had one small request: could you make the new ``specular_reflectance`` parameter `nullptr` by default (analogous to the changes here: https://github.com/mitsuba-renderer/mitsuba2/commit/a3531ac4245cb60017f45b4eb67a4522ee5c5d1e). It is a non-physical parameter (specular reflectance is determined by the Fresnel term, anything else is suspect and should  probably modeled in another way :-)), and I would prefer it to completely disappear from the system unless explicitly requested.

Thanks,
Wenzel

**Comment by jammm** [comment]
Hi @wjakob,

Thanks a lot for the feedback!
I made the changes you requested. Can you please take a look?

Cheers,
Aaryaman

File: src/bsdfs/roughconductor.cpp [hunk 1]
```diff
@@ -274,6 +277,10 @@ class RoughConductor final : public BSDF<Float, Spectrum> {
             F = fresnel_conductor(UnpolarizedSpectrum(dot(si.wi, m)), eta_c);
         }
 
+        /* If requested, include the specular reflectance component */
+        if (m_specular_reflectance)
+            F *= unpolarized<Spectrum>(m_specular_reflectance->eval(si, active));
```
Comment (wjakob) [inline src/bsdfs/roughconductor.cpp]: Change `weight` above to type `UnpolarizedSpectrum` & change this line to ` weight *= m_specular_reflectance->eval(si, active);
`.

File: src/bsdfs/roughconductor.cpp [hunk 2]
```diff
@@ -346,6 +353,10 @@ class RoughConductor final : public BSDF<Float, Spectrum> {
             F = fresnel_conductor(UnpolarizedSpectrum(dot(si.wi, H)), eta_c);
         }
 
+        /* If requested, include the specular reflectance component */
+        if (m_specular_reflectance)
+            F *= unpolarized<Spectrum>(m_specular_reflectance->eval(si, active));
```
Comment (wjakob) [inline src/bsdfs/roughconductor.cpp]: Change `result` above to type `UnpolarizedSpectrum` & change this line to ` result *= m_specular_reflectance->eval(si, active);
`.

**Comment by wjakob** [comment]
Two more small suggestions regarding polarization, the rest looks good.

**Comment by jammm** [comment]
Thanks !

I made the requested changes. Should I squash them all into a single commit?

**Comment by jammm** [comment]
Done! Please take a look :)

**Comment by wjakob** [comment]
Thanks, I can also do it :).

**Pull request merged**